### PR TITLE
Drop bluebird

### PR DESCRIPTION
**Pull request by proteria**
Node.js has great support for Promise for all versions supported by `npm-check-udpates`. This pull request removes the dependency for `bluebird`.

### CODE REVIEW

**Comment by raineorshine** [comment]
Thank you!

It looks like [finally is not supported](https://node.green/#ES2018-features-Promise-prototype-finally) on node v8, which explains the [build failures](https://travis-ci.org/tjunnone/npm-check-updates/jobs/552025002). We would have to find a solution to that before we can merge.

**Comment by raineorshine** [comment]
Great, thanks for your persistence. Any thoughts about the remaining errors on node 8? It still seems related to the `finally` statement. The test tries to delete temporary files as part of its cleanup process in the finally block.

**Comment by raineorshine** [comment]
@proteria Any ideas? Thanks!

**Comment by proteria** [comment]
The errors relating to deleting temporary files were caused by a race condition and fixed.

**Comment by raineorshine** [comment]
Thanks! Do you have time to look into the build error in node `v8`, or shall I?

**Comment by proteria** [comment]
That error only occasionally reproduces. I have no idea why it only happens on Node 8.

**Comment by stoically** [comment]
[`getAnalysis`](https://github.com/tjunnone/npm-check-updates/pull/550/files#diff-1415f5139deb67a37325f1bde416bcbdR437) sometimes wins the race with node8. My guess is that older node/v8's handled putting promises / timeouts in the event queue differently. A possible way to solve this would be to instantly resolve the setTimeout, something along the lines of https://github.com/stoically/npm-check-updates/commit/f2e5af8700ad96c7c40538b07bef0df5759a2e40

File: lib/npm-check-updates.js [hunk 1]
```diff
@@ -406,34 +406,35 @@ async function run(options={}) {
     }
 
     let timeout;
-    let timeoutPromise = Promise.resolve();
-
+    let timeoutPromise = new Promise(r => r);
```
Comment (stoically) [inline lib/npm-check-updates.js]: I'm curious, is there a reason you prefer that compared to `Promise.resolve()`?

**Comment by raineorshine** [comment]
@proteria This does the trick: https://github.com/tjunnone/npm-check-updates/commit/bc36bb0e64fd659b41b22c05ee0552b04d994c3a

I will merge the PR and then cherry-pick bc36bb

**Comment by stoically** [comment]
I doubt that helps - if anything passing in the actual package.json data makes `getAnalysis` even faster, so more likely to win the race against the `setTimeout`.

**Comment by raineorshine** [comment]
@stoically Actually, I tested it on node v8 and it works. The package.json is large enough that it reliably takes more than 1ms to resolve, not to mention the fs read.

**Comment by stoically** [comment]
Ah, got it, you make `getAnalysis` slower by passing in more stuff to process. Didn't thought about doing it that way, nice.

**Pull request merged**